actix-http/src/body.rs

```diff
@@ -589,4 +607,45 @@ mod tests {
             BodySize::Sized(25)
         );
     }
+
+    mod body_stream {
+        use super::*;
+
+        #[actix_rt::test]
+        async fn skips_empty_chunks() {
+            let body = BodyStream::new(stream::iter(
+                ["1", "", "2"]
+                    .iter()
+                    .map(|&v| Ok(Bytes::from(v)) as Result<Bytes, ()>),
+            ));
+            assert_eq!(read_all(body).await.unwrap(), Bytes::from("12"));
+        }
+    }
+
+    mod sized_stream {
+        use super::*;
+
+        #[actix_rt::test]
+        async fn skips_empty_chunks() {
+            let body = SizedStream::new(
+                2,
+                stream::iter(
+                    ["1", "", "2"]
+                        .iter()
+                        .map(|&v| Ok(Bytes::from(v))),
+                ),
+            );
+            assert_eq!(read_all(body).await.unwrap(), Bytes::from("12"));
```

Comment:
These tests pass with the old code that doesnt filter empty chunks.

It's clearer and more explicit to write the tests for this as test_vec, using manual polling and output checking. Eg:

```rust
assert_eq!(
    poll_fn(|cx| body.poll_next(cx)).await.unwrap().ok(),
    Some(Bytes::from("1"))
);
assert_eq!(
    poll_fn(|cx| body.poll_next(cx)).await.unwrap().ok(),
    Some(Bytes::from("2"))
);
```

Reply:
🤦‍♂️

Thanks!